app/src/main/java/com/duckduckgo/app/global/DuckDuckGoApplication.kt

```diff
@@ -83,11 +87,18 @@ open class DuckDuckGoApplication : HasActivityInjector, HasServiceInjector, HasS
         initializeStatistics()
         loadTrackerData()
         configureDataDownloader()
+        recordInstallationTimestamp()
 
         migrateLegacyDb()
         notificationRegistrar.registerApp()
     }
 
+    private fun recordInstallationTimestamp() {
+        if (!appInstallStore.hasInstallTimestampRecorded()) {
+            appInstallStore.installTimestamp = System.currentTimeMillis()
```

Comment:
For existing users, `installation timestamp` will actually mean time they upgraded to this app version, which is admittedly not the best name for that.

The need for the timestamp is that we want to be able to ensure they have been using the app for a minimum amount of time before showing them the default browser prompt. I don't mind that we'll have a timestamp for both new installs and upgrades. Otherwise, it will tricky to differentiate when to generate it as we don't have another way of detecting a new install vs an upgrade. At least not a way springing to mind - you have a plan for detecting this scenario?